### DISCUSSION TITLE
Check if device supports AR

### DISCUSSION THREAD

**Discussion by burungiu** [google/model-viewer#1913] [Q&A]
Hello, guys

I developed a React.JS web app to display a building in WebAR
There is a way to show "Your device doesn't render AR content" if device doesn't support it?

I tried to use the "ar-status" attribute of "model-viewer" element, but it is always "undefined". On Safari iOS it returns "undefined" constantly, but actually Safari iOS renders perfectly AR 

- model-viewer: latest, added in index.html as     
```
<script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>
<script nomodule src="https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js"></script>
```

Project could be reached at https://spital.imipasadegorj.ro/ (please be patient, it could load in 20-30 seconds or more)

- [x] Safari

- [x] iOS

**Comment by elalish** [comment]
We do our best to take care of this for you when using model-viewer (and this has just been significantly improved, so stay tuned for the next release), as we try to show and hide the button based on our heuristics. But they are only that; often it will show the button only to hide it after it fails, since it can only determine for sure with a user gesture.

**Comment by burungiu** [comment]
Ok, so, there is a way to detect if AR render is available on current device? I should release a project in few days and this is actually the last problem. 

Thank you very much for response!

**Comment by elalish** [comment]
Yeah, it's annoying for us too, but it's a privacy fingerprinting risk for the browser to tell you too much about the user's phone. In addition, you can't tell what apps are installed (for intenting out to sceneViewer for instance), so you can only find out when the user tries.

**Comment by jeanieherold** [comment]
Same here - We are about to launch a project but I am trying to find out anything to target regarding the user's device ability with AR but nothing is consistent with browsers to be a solid solution. Especially with how tight fingerprinting is being protected. If I could AT LEAST give the user an alert when ar-status fails that would help a little. I can't seem to find the proper way to target the ar-status.

This is where I am targeting model-viewer for a reset button on the page and you can see where the console.log is where I am trying to see if I can grab the ar status anytime I click that button either before or after i have started ar mode. It is always returning undefined even after I have used the ar mode and thinking it should have a value by then.

Can you help me with proper way to detect that status if it comes back 'failed' ? 

`    
    $(function () {
        const modelViewer = document.querySelector('#opal-mv');

        $('#ar-reset').on('click', function () {
            modelViewer.cameraOrbit = "0deg 75deg 1m";
            modelViewer.fieldOfView = "44.99999999999999";
            console.log(modelViewer.arStatus);
        });
    });
`

**Comment by elalish** [comment]
I just fixed this in #1475, and I'm working on putting a v1.2 release out today, hopefully. This enables an `ar-status` value of "failed". It can be a little confusing, as our documentation is staged off master, rather than the last release.

**Comment by jeanieherold** [comment]
Whatever happened with that last update you did broke being able to use on iphone with current setup. Was previously working fine. Now I get a message that says 'object could not be opened' (Which is nice :) )

This is a disaster though... We were going to launch today. 

Is there a setting I need to change for ios-src?

**Comment by elalish** [comment]
Standby; getting a fix out now!

**Comment by jeanieherold** [comment]
@elalish thank you!! and I made note that it works in the other Issue and also asked about how to version the cdn but I see that answer now in the tree/master/ repo (I cant find where I asked that in the other Issue - it might have been closed or combined.. i dunno. 

Again.. thank you for all you do!! :)
